code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -526,15 +526,14 @@ func (sc *StorageSmartContract) verifyChallenge(t *transaction.Transaction,
 	if last := allocChallenges.LatestCompletedChallenge; last != nil {
 		latestCompletedChallTime = last.Created
 	}
-
+	blobAlloc.LatestCompletedChallTime = latestCompletedChallTime
```

Comment:
We should not update blobAlloc's latest completed time here, it should be updated when the challenge is passed.

Reply:
Should we update it if its partially passed ?